_posts/2016-11-09-Rust-1.13.md

```diff
@@ -0,0 +1,469 @@
+---
+layout: post
+title: "Announcing Rust 1.13.0"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.13.0. Rust is
+a systems programming language focused on safety, speed, and concurrency.
+
+As always, you can [install Rust 1.13.0][install] from the appropriate page on
+our website, and check out the [detailed release notes for 1.13.0][notes] on
+GitHub. 1448 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: 
+
+### What's in 1.13.0 stable
+
+The two big things in Rust 1.13 stable are compiler performance and a new
+operator, `?`. As usual, there's also a number of library stabilizations and
+other minor improvements and changes.
+
+#### The `?` operator
+
+Rust has gained a new operator, `?`, that's used to make error-handling more
+pleasant.
+
+Imagine we had some code to read some data from a file:
+
+```rust
+fn read_username_from_file() -> Result<String, io::Error> {
+    let f = File::open("username.txt");
+
+    let mut f = match f {
+        Ok(file) => file,
+        Err(e) => return Err(e),
+    };
```

Comment:
I tried doing so, but this particular example is too simple, and doesn't look like a compelling justification of `?` over `try!`. I have a more complex example [here](https://gist.github.com/brson/1beda7c9eef6c6d1d33e3ba587da21af) in comparison.

I do like that this current explains what `?` does, not just that it's the same as `try!`.